oioioi/contests/utils.py

```diff
@@ -247,6 +256,96 @@ def visible_rounds(request):
     return [r for r in queryset if controller.can_see_round(request, r)]
 
 
+@request_cached
+def get_number_of_rounds(request):
+    """Returns the number of rounds in the current contest.
+    """ 
+    return Round.objects.filter(contest=request.contest).count()
+
+
+def get_contest_dates(request):
+    """Returns the end_date of the latest round and the start_date
+    of the earliest round.
+    """ 
+    if isinstance(request, HttpRequest):
+        rtimes = rounds_times(request, request.contest)
+    else:
+        rtimes = generic_rounds_times(None, request.contest)
+
+    ends = [
+        rt.get_end()
+        for rt in rtimes.values()
+    ]
+    starts = [
+        rt.get_start()
+        for rt in rtimes.values()
+    ]
+
+    if (not starts or starts.count(None)):
+        min_start = -1
+    else:
+        min_start = min(starts)
+
+    if (not ends or ends.count(None)):
+        max_end = -1
+    else:
+        max_end = max(ends)
```

Comment:
I'd say
```py
if starts and None not in starts:
    min_start = min(starts)
else:
    min_start = -1
```
To me, this is more self explanatory.

Reply:
You're right, this seems much better. Fixed!